client/src/services/DonuService.ts

```diff
@@ -1,22 +1,19 @@
 import { ModelInterface } from '@/types/types';
 import * as Donu from '@/types/typesDonu';
 import { donuToModel } from '@/utils/DonuUtil';
+import { postUtilMem } from '@/utils/FetchUtil';
 
 /** Send the request to Donu */
-async function callDonu (request: Donu.Request): Promise<Donu.Response> {
+const callDonu = (request: Donu.Request): Promise<Donu.Response> => {
   try {
-    const response = await fetch(process.env.DONU_ENDPOINT, {
-      body: JSON.stringify(request),
-      method: 'POST',
-    });
-    return response.json() as Donu.Response;
+    return postUtilMem(process.env.DONU_ENDPOINT, request);
```

Comment:
so how this thing works is that instead of storing data for the saved runs inside the components, it calls the fetch for all runs including all the saved runs every time and relies on the cache to ensure that only the query for the new run is sent to donu. i suppose i could remove the cache and just fetch all runs every time, that should work too, could have slightly more latency though.

Reply:
From what I understand our HMI should be kept as lean as possible.
I think it would be better to not cache anything and do multiple calls to DONU API.
Worth talking to @adamocarolli about it in [#aske-architecture](https://unchartedsoftware.slack.com/archives/C01SJ24GD1C)

Reply:
Right, the client's browser cache wont handle this POST call (the server should have a shared cache though so long-running executions are avoided).

@mj3cheun I believe the cache you are using is a Map that does not free memory and does not get freed across pages. You may want to consider building a `free` memory policy. However, I agree you do not want to be making several past run calls here as the server-side computation is very heavy for these calls.

Reply:
ok for now i removed the memoization, i will look into seeing what is the best way to free memory under conditions such as on url change or something